### PR TITLE
potential buffer overflow vulnerability

### PR DESCRIPTION
**Pull request by bashrc-real**
replaced fscanf with ifstream. Also moved the static initialization out of the loop and changed it to const. static variables makes the function not thread safe.

### CODE REVIEW

**Comment by bertmaher** [comment]
Thanks!  I will merge this today.

File: libredex/PgoFiles.cpp [hunk 1]
```diff
@@ -21,26 +21,25 @@
...
+  if (!input.isOpen()){
     return std::vector<std::string>();
   }
-  std::vector<std::string> coldstart_classes;
-  char buf[kMaxLineLength];
-  buf[0] = 'L';
-  while (fscanf(fp, "%s", buf + 1) == 1) {
-    static int lentail = strlen(kClassTail);
-    std::string clzname(buf);
+  string clzname;
```
Comment (benjamin-bader) [inline libredex/PgoFiles.cpp]: looks like `string` is missing a `std::` prefix; will this compile without it?

Comment (bashrc-real) [inline libredex/PgoFiles.cpp]: Fixed. Had missed it in last minute changes.

File: libredex/PgoFiles.cpp [hunk 2]
```diff
@@ -21,26 +21,25 @@
...
+  const int lentail = strlen(kClassTail);
   const int kMaxLineLength = 1024;
   auto file = m_coldstart_class_filename.c_str();
-  FILE* fp = fopen(file, "r");
-  if (fp == nullptr) {
+  
+  std::vector<std::string> coldstart_classes;
+  
+  std::ifstream input(file);
+  if (!input.isOpen()){
```
Comment (bertmaher) [inline libredex/PgoFiles.cpp]: isOpen isn't a method of input (maybe you meant is_open).  But I'd recommend just using operator bool and saying `if (!input)`

File: libredex/PgoFiles.cpp [hunk 3]
```diff
@@ -21,26 +21,25 @@
  */
 std::vector<std::string> PgoFiles::load_coldstart_classes() {
   const char* kClassTail = ".class";
+  const int lentail = strlen(kClassTail);
   const int kMaxLineLength = 1024;
```
Comment (bertmaher) [inline libredex/PgoFiles.cpp]: Remove kMaxLineLength, since it's unused now.

File: libredex/PgoFiles.cpp (hunk 2 above)
Comment (bashrc-real) [inline libredex/PgoFiles.cpp]: fixed in latest commit

**Pull request closed without merging**